src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -26,11 +32,47 @@ public static function onShowSearchHitTitle(
 		array &$query,
 		array &$attributes
 	): void {
-		if ( $title->getNamespace() !== WB_NS_ITEM ) {
+		$itemId = WikibaseRepo::getEntityIdLookup()->getEntityIdForTitle( $title );
+
+		if ( !( $itemId instanceof ItemId ) ) {
 			return;
 		}
 
-		// TODO: get item site link and replace $title
+		$pageTitle = WikibaseFacetedSearchExtension::getInstance()->getItemPageLookup()->getPageTitle( $itemId );
+
+		if ( $pageTitle === null ) {
+			return;
+		}
+
+		if ( $result instanceof EntityResult ) {
+			CirrusShowSearchHitHandler::onShowSearchHitTitle(
+				$title,
+				$titleSnippet,
+				$result,
+				$terms,
+				$specialSearch,
+				$query,
+				$attributes
+			);
+		} else {
+			self::runHtmlPageLinkRendererEndHook( $title, $titleSnippet, $attributes );
+		}
```

Comment:
When replacing $title, the link gets the text from the new Title object. We want to keep the original Item title, so we have to let the original link text builders run.